OpenRA.Game/Server/Server.cs

```diff
@@ -659,37 +668,51 @@ void CompleteConnection()
 							Log.Write("server", ex.ToString());
 						}
 
-						events.Add(new CallbackEvent(() =>
+						var profileBlocked = false;
+						var profileAdmin = false;
+
+						if (profile != null && !string.IsNullOrEmpty(Settings.ProfileIDFilterURL))
 						{
-							var notAuthenticated = Type == ServerType.Dedicated && profile == null && (Settings.RequireAuthentication || Settings.ProfileIDWhitelist.Length > 0);
-							var blacklisted = Type == ServerType.Dedicated && profile != null && Settings.ProfileIDBlacklist.Contains(profile.ProfileID);
-							var notWhitelisted = Type == ServerType.Dedicated && Settings.ProfileIDWhitelist.Length > 0 &&
-								(profile == null || !Settings.ProfileIDWhitelist.Contains(profile.ProfileID));
+							try
+							{
+								var httpClient = HttpClientFactory.Create();
+								var url = Settings.ProfileIDFilterURL + profile.ProfileID;
+								var httpResponseMessage = await httpClient.GetAsync(url);
+								var result = await httpResponseMessage.Content.ReadAsStringAsync();
+								profileBlocked = result == "block";
+								profileAdmin = result == "admin";
```

Comment:
Wouldn't it be simpler to let the user define a list of admin IDs in the launch script for the specific server? Or is the idea to have the same admins for all servers?

Reply:
There is only one URL, so it isn't completely straight-forward to keep the community curated banlist while having a custom admin list.

Reply:
A server host can define their own script that hardcodes the list of admins but still fetches a community block list.